tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,194 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
+from ocs_ci.utility.deployment_openshift_logging import (
+    check_health_of_clusterlogging, get_clusterlogging_subscription
+)
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.framework.pytest_customization.marks import (
+    post_ocp_upgrade
+)
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.utility.utils import (
+    TimeoutSampler, get_ocp_version, run_cmd
+)
+
+logger = logging.getLogger(__name__)
+
+
+def check_cluster_logging():
+    """
+    The function has some pre-checks before
+    the openshift-logging upgrade
+
+    """
+
+    assert get_all_pods(
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    ), 'Some or all pods missing in namespace'
+
+    assert check_health_of_clusterlogging(), "Cluster is not Healthy"
+
+
+def check_csv_logging_upgrade_phase():
+    """
+    The function checks in the CSV for the phase "Succeeded"
+    for cluster-logging operator and Elastic-search
+    operator
+
+    """
+
+    csv = ocp.OCP(
+        kind=constants.CLUSTER_SERVICE_VERSION,
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    )
+    csv_after_upgrade = csv.get(out_yaml_format=True)
+    clo_phase_after_upgrade = (
+        csv_after_upgrade.get('items')[0].get('status').get('phase')
+    )
+    eso_phase_after_upgrade = (
+        csv_after_upgrade.get('items')[1].get('status').get('phase')
+    )
+    if clo_phase_after_upgrade and eso_phase_after_upgrade == 'Succeeded':
+        logger.info("CSV phase Succeeded")
+        return True
+
+    logger.debug("Upgrade not completed yet")
+    return False
+
+
+def check_version_after_upgrade():
+    """
+    Checks the version after upgrade
+
+    """
+    csv = ocp.OCP(
+        kind=constants.CLUSTER_SERVICE_VERSION,
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    )
+    csv_after_upgrade = csv.get(out_yaml_format=True, wait=30)
+    clo_version_after_upgrade = (
+        csv_after_upgrade.get('items')[0].get('spec').get('version')
+    )
+    eso_version_after_upgrade = (
+        csv_after_upgrade.get('items')[1].get('spec').get('version')
+    )
+    if config.UPGRADE['upgrade_logging_channel'] in (
+        clo_version_after_upgrade and eso_version_after_upgrade
+    ):
+        logger.info(
+            f"Upgraded version of Cluster-logging "
+            f"operator {clo_version_after_upgrade}"
+        )
+        logger.info(
+            f"Upgraded version of Elastic-search "
+            f"operator {eso_version_after_upgrade}"
+        )
+        return True
+    return False
+
+
+@post_ocp_upgrade
+@pytest.mark.polarion_id("OCS-2201")
+class TestUpgradeLoggingAfterOCPUpgrade():
```

Comment:
Can logging upgrade be performed without upgrading OCP  first?

Reply:
No logging upgrade cannot be performed before the OCP upgrade, Hence i used the decorator @post_ocp_upgrade. is it not enough ?

Reply:
using `post_ocp_upgrade` marker is ok, question was about Class name
the test itself does not trigger ocp upgrade, marker means that it will run if this marker will be selected as run-ci parameter

Reply:
done